netlogo-core/src/main/shape/LinkShape.scala

```diff
@@ -211,7 +211,11 @@ class LinkShape extends BaseLinkShape with Cloneable with java.io.Serializable w
     do {
       split(line, left, right)
       line = left
-    } while (!dest.contains(line.getP2))
+    } while (!dest.contains(line.getP2) &&
+      Math.abs(line.getP1.distance(line.getP2)) > 0.25) // do not split indefinitely.
```

Comment:
Yeah, pretty sure this was only happening on the edge cases of [the definition of Insideness](https://docs.oracle.com/javase/8/docs/api/java/awt/Shape.html#def_insideness).